server/util/parseUserMentions.ts

```diff
@@ -0,0 +1,27 @@
+export const parseUserMentions = (text): any[] => {
+  // Extract links to Commonwealth profiles, so they can be processed by the server as mentions
+  if (!text) return [];
+  try {
+    const parsedText = JSON.parse(text);
+    return (parsedText.ops || [])
+      .filter((op) => {
+        return op.attributes?.link?.length > 0 && typeof op.insert === 'string' && op.insert?.slice(0, 1) === '@';
+      })
+      .map((op) => {
+        const chunks = op.attributes.link.split('/');
+        const refIdx = chunks.indexOf('account');
+        return [chunks[refIdx - 1], chunks[refIdx + 1]];
+      });
+  } catch (e) {
+    const regexp = RegExp('\\[\\@.+?\\]\\(.+?\\)', 'g');
+    const matches = text.match(regexp);
+    if (matches && matches.length > 0) {
+      return matches.map((match) => {
+        const chunks = match.slice(0, match.length - 1).split('/');
+        const refIdx = chunks.indexOf('account');
+        return [chunks[refIdx - 1], chunks[refIdx + 1]];
+      });
+    }
```

Comment:
- This looks somewhat brittle. If the second stage of mentions parsing fails, any backend route will return a 500 error and be nonfunctional. Can we at least add a nested try-catch to that part of the logic (for both Markdown and Quill) so if unpacking chunks fails, we don't get a 500 error?
- Can we add a check that the mention link matches the server URL?

Reply:
Actually never mind on the second part, looks like these are relative links.

Reply:
Hmmm I see what you mean, lemme go back in

Reply:
Alright, all set

Reply:
This will still result in a 500 error - I was thinking we could catch the cases where these might fail?

```
    return (parsedText.ops || [])
      .filter((op) => {
        return op.attributes?.link?.length > 0 && typeof op.insert === 'string' && op.insert?.slice(0, 1) === '@';
      })
      .map((op) => {
        const chunks = op.attributes.link.split('/');
        const refIdx = chunks.indexOf('account');
        return [chunks[refIdx - 1], chunks[refIdx + 1]];
      });
```

```
      return matches.map((match) => {
        const chunks = match.slice(0, match.length - 1).split('/');
        const refIdx = chunks.indexOf('account');
        return [chunks[refIdx - 1], chunks[refIdx + 1]];
      });
```

Reply:
I guess I'm a little confused... What is the situation you're hitting 500 errors for? AFAICT, the text argument to parseUserMentions is always going to be a string (even if you bypass our controller methods and make a direct API request modifying body), so it's hard for me to imagine the error scenario. (And I can't get one to throw in practice either.) My mental model, and (so far in my QAing) the behavior regardless of what kind of comment/thread body I pass,  is that this function will, at worst, return an empty array.

Reply:
Hmmm.... In that case (refIdx === -1) I think the endpoint will take array returned by parseUserMentions and try to fetch an Address entry for each entry—any improper array elements will end up as a null value (since no Address will be found if the indexing isn't right), and that null value gets filtered from user mentions pretty immediately. Also, indexing into an array (`chunks`) with -2 should just return undefined, so it wouldn't be solved by try/catch wrapping anyway, right?

Definitely game to make this less brittle I'm just a little unsure how to intervene since I haven't been able to throw an error yet when passing different types of invalid mentions